SwiftGit2Tests/RepositorySpec.swift

```diff
@@ -29,6 +29,99 @@ class RepositorySpec: QuickSpec {
 				)))
 			}
 		}
+
+		describe("Reposityory.Type.clone()") {
+			it("should handle local clones") {
+				let remoteRepo = Fixtures.simpleRepository
+				let localURL = self.temporaryURLForPurpose("local-clone")
+				let result = Repository.cloneFromURL(remoteRepo.directoryURL!, toWorkingDirectory: localURL, localClone: true)
+
+				expect(result).to(haveSucceeded())
+
+				if case .Success(let clonedRepo) = result {
+					expect(clonedRepo.directoryURL).notTo(beNil())
+				}
+			}
+
+			it("should handle bare clones") {
+				let remoteRepo = Fixtures.simpleRepository
+				let localURL = self.temporaryURLForPurpose("bare-clone")
+				let result = Repository.cloneFromURL(remoteRepo.directoryURL!, toWorkingDirectory: localURL, localClone: true, bare: true)
+
+				expect(result).to(haveSucceeded())
+
+				if case .Success(let clonedRepo) = result {
+					expect(clonedRepo.directoryURL).to(beNil())
+				}
+			}
+
+			it("should have set a valid remote url") {
+				let remoteRepo = Fixtures.simpleRepository
+				let localURL = self.temporaryURLForPurpose("valid-remote-clone")
+				let cloneResult = Repository.cloneFromURL(remoteRepo.directoryURL!, toWorkingDirectory: localURL, localClone: true)
+
+				expect(cloneResult).to(haveSucceeded())
+
+				if case .Success(let clonedRepo) = cloneResult {
+					let remoteResult = clonedRepo.remoteWithName("origin")
+					expect(remoteResult).to(haveSucceeded())
+
+					if case .Success(let remote) = remoteResult {
+						expect(remote.URL).to(equal(remoteRepo.directoryURL?.absoluteString))
+					}
+				}
+			}
+
+			it("should be able to clone a remote repository") {
+				let remoteRepoURL = NSURL(string: "https://github.com/libgit2/libgit2.github.com.git")
+				let localURL =  self.temporaryURLForPurpose("public-remote-clone")
+				let cloneResult = Repository.cloneFromURL(remoteRepoURL!, toWorkingDirectory: localURL)
+
+				expect(cloneResult).to(haveSucceeded())
+
+				if case .Success(let clonedRepo) = cloneResult {
+					let remoteResult = clonedRepo.remoteWithName("origin")
+					expect(remoteResult).to(haveSucceeded())
+
+					if case .Success(let remote) = remoteResult {
+						expect(remote.URL).to(equal(remoteRepoURL?.absoluteString))
+					}
+				}
+			}
+
+			let privateRepo = NSProcessInfo.processInfo().environment["SG2TestPrivateRepo"]
+			let username = NSProcessInfo.processInfo().environment["SG2TestUsername"]
+			let publicKey = NSProcessInfo.processInfo().environment["SG2TestPublicKey"]
+			let privateKey = NSProcessInfo.processInfo().environment["SG2TestPrivateKey"]
+			let passphrase = NSProcessInfo.processInfo().environment["SG2TestPassphrase"]
```

Comment:
Can you extract the `environment` to a local variable since we're using it several times? IIRC, it's quite expensive to get, which could affect the runtime of the unit tests.